packages/vaadin-template-renderer/src/vaadin-template-renderer.js

```diff
@@ -0,0 +1,44 @@
+import './vaadin-template-renderer-templatizer.js';
+import { Templatizer } from './vaadin-template-renderer-templatizer.js';
+
+function createRenderer(template) {
+  const templatizer = Templatizer.create(template);
+
+  return (root) => {
+    template.__templatizer = templatizer;
+    template.__templatizer.render(root);
+  };
+}
+
+function processTemplate(component, template) {
+  component.renderer = createRenderer(template);
+}
+
+function processTemplates(component, children) {
+  [...children]
+    .filter((child) => {
+      return child instanceof HTMLTemplateElement;
+    })
+    .forEach((template) => {
+      // Ignore templates which have been processed earlier
+      if (template.__templatizer) {
+        return;
+      }
+
+      processTemplate(component, template);
+    });
+}
+
+const observer = new MutationObserver((mutations) => {
+  mutations.forEach(({ target, addedNodes }) => {
+    processTemplates(target, addedNodes);
+  });
+});
+
+window.Vaadin = window.Vaadin || {};
```

Comment:
Regarding [the discussion](https://github.com/vaadin/web-components/issues/256#issuecomment-828323324), probably, a deprecated warning should be added here as well as a configuration property to suppress it.